### ISSUE TITLE
NugetAuthenticate and PowershellGet Nuget provider

### ISSUE THREAD

**Issue by dpurge** [microsoft/azure-pipelines-tasks#13027]
*Type*: Question

**Enter Task Name**: NuGetAuthenticate

## Environment
- Server - on-premises - version Dev17.M153.5
- Agent - Private - version 2.153.1
- Powershell version 5.1.17763.1007
- OS - Windows Server 2019 Standard, build 17763.1217

## Issue Description

We have started to use Azure Artifacts and needed a way to authenticate with the feeds.

I have built NuGetAuthenticate locally and installed it on our on-premises instance. It works perfectly in all places where we call nuget.exe directly or use provided nuget tasks.

But we also have a custom task that installs powershell modules. The concept is the same that exists for installing build tools. We create a temporary directory, add it to PsModulePath, call Register-PsRepository if it has not been registered yet and then we use Save-Module to place the module in that directory. The details of the logic do not matter that much, it's been working for a couple of years now with the preauthorized feed url. 

I hoped that Powershell nuget provider would work with nuget credential provider which is installed by NuGetAuthenticate, but when I try to install modules after running NuGetAuthenticate I am getting an error that says that I need to provide credential to access the feed. It means that it does not work for installing PsModules.

My question: do you see a way of making NuGetAuthenticate task to install credential provider in such a way that PowershellGet nuget provider can use it? Or maybe there is a way for me to create System.Management.Automation.PSCredential object when we run under the build by extracting username and PAT from the service connection?

### Error logs

This is the error that I get from Save-Module cmdlet:

System.Exception: Cannot access 'https://pkgs.dev.azure.com/...snip.../nuget/v3/index.json'. Are you missing 'Credential' parameter in the cmdlet?

**Comment by apluche** [comment]
Hi @dpurge , 

This feature does not currently exist. Could you create a suggestion ticket in this repo?

**Comment by dpurge** [comment]
Hi @apluche, thank you for your response. Apologies for the delay in my answer, I was trying to find instructions how to create a "suggestion ticket" in this repo, could not find either instructions or an example and later was a bit busy :)

Should I just create a new ticket and explain what I would expect and why I think it's useful?
It would be a copy of this question - so should I do that?

FYI - since the functionality did not exist, I have already rewritten our task. Now I simply call nuget.exe to download the package, so that we can use credential providers installed by NuGetAuthenticate. It works, therefore I no longer have that much interest in this functionality, but still think Azure Artifacts not working for hosting powershell modules "out of the box" is not ideal :)

Since you have answered my question, I think it's right to close the ticket.

**Comment by apluche** [comment]
Hi @dpurge , a suggestion ticket is the same as an issue but they type that is assigned to it is different. I'm glad to hear that you were able to unblock yourself. If you would like Azure Artifacts to consider this feature, please make sure to create a ticket since we keep track of those in our backlog.

**Comment by dpurge** [comment]
Created suggestion for a new feature: #13215
